### PR TITLE
ISPN-12150 Improve CLI server report to work with MacOs

### PR DESCRIPTION
**Pull request by gustavolira**
https://issues.redhat.com/browse/ISPN-12150

### CODE REVIEW

File: server/runtime/src/main/java/org/infinispan/server/Server.java [hunk 1]
```diff
@@ -554,14 +554,15 @@ public TaskManager getTaskManager() {
...
-      if (os != OS.LINUX) {
+      if (os == OS.WINDOWS) {
          return CompletableFutures.completedExceptionFuture(log.serverReportUnavailable(os));
       }
       long pid = ProcessInfo.getInstance().getPid();
       Path home = serverHome.toPath();
       Path root = serverRoot.toPath();
       ProcessBuilder builder = new ProcessBuilder();
-      builder.command("sh", "-c", home.resolve("bin/report.sh").toString(), Long.toString(pid), root.toString());
+      String reportFile = String.format("bin/%s", OS.getCurrentOs() == OS.LINUX ? "report.sh" : "report-osx.sh");
```
Comment (tristantarrant) [inline server/runtime/src/main/java/org/infinispan/server/Server.java:564]: I'd rather have a switch statement with explicit os-specific calls.

Comment (gustavolira) [reply server/runtime/src/main/java/org/infinispan/server/Server.java]: @tristantarrant updated

**Comment by tristantarrant** [comment]
@gustavonalle @karesti could you please test this on OSX ?

**Comment by gustavonalle** [comment]
Testing...

**Comment by gustavonalle** [comment]
The report is generated, and it seems correct. Only weird thing I found is the file ```thread-dump``` is empty

**Comment by gustavolira** [comment]
> The report is generated, and it seems correct. Only weird thing I found is the file `thread-dump` is empty. Not sure if it's a problem on Linux as well.
> 
> `http://127.0.0.1:11222/rest/v2/server/threads` correctly generates the dump on MacOS btw

IIRC we have this same behaviour with Linux

**Comment by gustavonalle** [comment]
Merged, thanks @gustavolira

**Pull request merged**